src/native/mac/qgsmacnative.mm

```diff
@@ -0,0 +1,36 @@
+/***************************************************************************
+    qgsmacnative.cpp - abstracted interface to native Mac objective-c
+                             -------------------
+    begin                : January 2014
+    copyright            : (C) 2014 by Larry Shaffer
+    email                : larrys at dakotacarto dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmacnative.h"
+
+#include <Cocoa/Cocoa.h>
+
+QgsMacNative::~QgsMacNative()
+{
+}
+
+const char* QgsMacNative::currentAppLocalizedName()
+{
+  return [[[NSRunningApplication currentApplication] localizedName] UTF8String];
+}
+
+void QgsMacNative::currentAppActivateIgnoringOtherApps()
+{
+  // valid for Mac OS X >= 10.6
+  [[NSRunningApplication currentApplication] activateWithOptions:
+    (NSApplicationActivateAllWindows | NSApplicationActivateIgnoringOtherApps)];
```

Comment:
Can test with instance where app windows are not all forefront:

* open main QGIS app window
* open Composer window
* switch to other app, but not one that covers the whole screen, so you can still see one of the QGIS windows
* click on a QGIS window, this will bring it forward, but not all QGIS windows, e.g. maybe the Composer window is still hidden
* choose `Window -> Bring all to Front` menu action, which will bring all QGIS windows to forefront, now in front of any other apps that previous hid some windows

In older versions of Mac OS and OS X, clicking on any window of another app would always bring all its windows to the forefront. Once Apple changed this functionality across the OS, `Window -> Bring all to Front` became a standard GUI construct.